internal/container/default_registry.go

```diff
@@ -8,31 +8,67 @@ import (
 	"github.com/pkg/errors"
 )
 
-// The Host of a container registry where we can push images.
-// ex)
-// localhost:32000
-// gcr.io/windmill-public-containers
-type Registry string
-
 var escapeRegex = regexp.MustCompile(`[/:@]`)
 
 func escapeName(s string) string {
 	return string(escapeRegex.ReplaceAll([]byte(s), []byte("_")))
 }
 
-// Produces a new image name that is in the specified registry.
+type Registry struct {
+	// The Host of a container registry where we can push images. e.g.:
+	//   - localhost:32000
+	//   - gcr.io/windmill-public-containers
+	PushHost string
```

Comment:
are you sure you don't want to keep this as HostFromLocal vs HostFromCluster?

PushHost vs PullHost isn't quite accurate - for example, when you have image dependencies, you'll both push and pull from the PushHost